satpy/readers/file_handlers.py

```diff
@@ -31,10 +30,7 @@ class BaseFileHandler(metaclass=ABCMeta):
 
     def __init__(self, filename, filename_info, filetype_info):
         """Initialize file handler."""
-        if isinstance(filename, PurePath):
-            self.filename = str(filename)
-        else:
-            self.filename = filename
+        self.filename = filename
```

Comment:
This change will break pathlib objects for all other readers right? (assuming the low-level I/O library doesn't support them)

Reply:
That said, I'm ok with this.

Reply:
probably yes... Is this bad?

Reply:
I think @gerritholl was the first user to point out that Satpy didn't work with pathlib objects so he should maybe make the final decision. It probably isn't great that this breaks it for all other readers, but I'm not sure how many readers need strings for their lower-level I/O libraries either.